### PR TITLE
[SMALLFIX] Reduce thread usage in BlockMasterSync

### PR DESCRIPTION
**Pull request by aaudiber**
Before we would execute one executor task per block. Now we will perform at most one task per heartbeat. This could be relevant if hundreds of blocks are freed during a single heartbeat.

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java [hunk 1]
```diff
@@ -214,55 +219,38 @@ private void handleMasterCommand(Command cmd) throws IOException, ConnectionFail
...
     @Override
     public void run() {
-      boolean success = false;
-      try {
-        mBlockWorker.removeBlock(mSessionId, mBlockId);
-        success = true;
-        synchronized (mRemovingBlockIdToFinished) {
-          mRemovingBlockIdToFinished.put(mBlockId, true);
-        }
-        LOG.info("Block {} removed at session {}", mBlockId, mSessionId);
```
Comment (yupeng9) [inline core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java]: Can you keep this logging, probably at debug level?

File: core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java [hunk 2]
```diff
@@ -214,55 +219,38 @@ private void handleMasterCommand(Command cmd) throws IOException, ConnectionFail
...
+        } catch (IOException e) {
+          LOG.warn("Failed master free block cmd for: {}.", block, e);
+        } catch (InvalidWorkerStateException e) {
+          LOG.warn("Failed master free block cmd for: {} due to block uncommitted.", block, e);
+        } catch (BlockDoesNotExistException e) {
+          LOG.warn("Failed master free block cmd for: {} due to block not found.", block, e);
         }
+        // The remove operation fails, so remove the block from the map in order to make it
+        // possible for another BlockRemover to remove it later.
+        mRemovingBlockIdToFinished.remove(block);
```
Comment (yupeng9) [inline core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java]: synchronize on mRemovingBlockIdToFinished ?

Comment (aaudiber) [reply core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java]: done, thanks

File: core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java (hunk 1 above)
Comment (aaudiber) [reply core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java]: done, thanks

**Pull request merged**